Problems/2630/ans_2630_JHP.py

```diff
@@ -0,0 +1,45 @@
+# https://www.acmicpc.net/problem/2630
+# 문제 : 정사각형 모양의 같은 색(0,1) 종이를 나누는 문제
+# 입력 : 첫째줄 = N, NxN의 정사각형
+# 출력 : 첫째 줄 = 하얀색 색종이, 둘째 줄 = 파란색 종이
+
+import sys
+
+def paper_div(x,y,n):
+    global one,zero
+    global array_n    
+    check=array_n[x][y]
+
+    for i in range(x,x+n):
+        for j in range(y,y+n):
+            if check!=array_n[i][j]:     #나머지 사각형의 색깔과 일치하지 않는다면 실행합니다.
+                paper_div(x,y,n//2)            #1사분면
+                paper_div(x,y+n//2,n//2)       #2사분면
+                paper_div(x+n//2,y,n//2)       #3사분면
+                paper_div(x+n//2,y+n//2,n//2)  #4사분면
```

Comment:
```suggestion
				zero = 0
				one = 0
                tempZero, tempOne = paper_div(x,y,n//2)            #1사분면
                zero += tempZero
                one += tempOne
                tempZero, tempOne = paper_div(x,y+n//2,n//2)       #2사분면
                zero += tempZero
                one += tempOne
                tempZero, tempOne = paper_div(x+n//2,y,n//2)       #3사분면
                zero += tempZero
                one += tempOne
                tempZero, tempOne = paper_div(x+n//2,y+n//2,n//2)  #4사분면
                zero += tempZero
                one += tempOne
                ...
                ...
                ...
	return zero, one
```
이런식이면 어떨지. global 안쓰고 처리

Reply:
잘봤습니다. 분할정복 이거 다음으로 1780번 풀어보면 좋을것 같아요
풀이법은 거의 유사함

Reply:
흠, 저거도 방법이네요...
global 선언을 해서 빈칸을 만들어놓고, 매번 거기에 채워넣는 형태에 익숙해져있는거 같아요.

그게 좀 비효율적이고, 함수의 input,output이 좀 불명확할 수 있는데 함수 내부 지역변수로 return을 하니 역할이 좀더 명확해지네요.

앞으로 풀 때 해당 부분도 고민해 보겠습니다. 좋은 의견 감사해요.